bindings/C/adios2/c/adios2_c_attribute.cpp

```diff
@@ -169,7 +169,7 @@ adios2_error adios2_attribute_data(void *data, size_t *size,
                 *size = attributeCpp->m_Elements;
                 char **dataT = reinterpret_cast<char **>(data);
 
-                for (auto e = 0; e < *size; ++e)
+                for (size_t e = 0; e < *size; ++e)
```

Comment:
Just a quick comment: `size` is going to become a worse variable name in the future, as C++17 introduces `std::size` and `using std::size` has gotten into various popular C++ libraries. Whether there will actually be a conflict, I don't know.

Reply:
@NAThompson thanks for pointing this out. We need to be in the lookout for new language features.